### PR TITLE
Move test fips configuration to script plugin

### PR DESCRIPTION
**Pull request by rjernst**
This commit moves the configuration of all test jvms for fips to a
script plugin. Fips testing is something very specific to the
Elasticsearch build and does not need to be passed on to plugin authors.

### CODE REVIEW

File: gradle/fips.gradle [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import org.elasticsearch.gradle.ExportElasticsearchBuildResourcesTask
+import org.elasticsearch.gradle.testclusters.ElasticsearchCluster
+import org.elasticsearch.gradle.testclusters.ElasticsearchCluster
+
+// Common config when running with a FIPS-140 runtime JVM
+if (Boolean.parseBoolean(System.getProperty("tests.fips.enabled"))) {
```
Comment (mark-vieira) [inline gradle/fips.gradle:6]: We should use `BuildParams.inFipsJvm` here instead.

**Review by mark-vieira — CHANGES_REQUESTED** [review]
On comment, also [this logic](https://github.com/elastic/elasticsearch/blob/2e1e33d91099c4e8fb324c448875a1a015829fd5/buildSrc/src/main/java/org/elasticsearch/gradle/ElasticsearchJavaPlugin.java#L274-L277) needs to get moved as well.

File: gradle/fips.gradle (hunk 1 above)
Comment (rjernst) [reply gradle/fips.gradle]: Done

**Comment by rjernst** [comment]
@elasticmachine run elasticsearch-ci/1

**Review by mark-vieira — APPROVED** [review]
👍

**Pull request merged**